src/modules/grubcfg/main.py

```diff
@@ -90,6 +90,12 @@ def modify_grub_default(partitions, root_mount_point, distributor):
     swap_outer_uuid = ""
     swap_outer_mappername = None
     no_save_default = False
+    unencrypted_separate_boot = False
+
+    for partition in partitions:
+        if (partition["mountPoint"] == "/boot"
+                and "luksMapperName" not in partition):
+            unencrypted_separate_boot = True
```

Comment:
Somehow this code just seems icky: what you're expressing is "is any partition an unencrypted boot partition". Python has `any()` for that (and C++ has `std::any_of()`) and the predicate is fairly simple to write .. but nothing I can come up with is much shorter than what you have here :(

Reply:
Yes, it is ugly. Also duplicated code from the initramfs module. I can see if I can come up with something better.

Reply:
Personally, I consider this standard iterative pattern no harder to read than the functional version. My C++ version in `dracutlukscfg` is almost the same, except that I actually break the loop after an unencrypted separate `/boot` is found, which is missing in the above Python code.

Reply:
I updated to use a generator expression instead to look more pretty